presto-product-tests/src/main/java/com/facebook/presto/tests/utils/JdbcDriverUtils.java

```diff
@@ -78,7 +78,11 @@ else if (usingTeradataJdbcDriver(connection)) {
 
     public static boolean usingPrestoJdbcDriver(Connection connection)
     {
-        return getClassNameForJdbcDriver(connection).equals("com.facebook.presto.jdbc.PrestoConnection");
+        String canonicalClassName = getClassNameForJdbcDriver(connection);
```

Comment:
Wrapping the connection in an anonymous inner class has the effect of making the canonical class name null. This is kind of a hack. I'm open to suggestions for making it less hacky.